modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/download/task/UpdateShowNotesTask.kt

```diff
@@ -21,24 +29,50 @@ class UpdateShowNotesTask @AssistedInject constructor(
     private val showNotesManager: ServerShowNotesManager
 ) : CoroutineWorker(context, params) {
     companion object {
+        private const val TASK_NAME = "UpdateShowNotesTask"
         const val INPUT_PODCAST_UUID = "podcast_uuid"
-        const val INPUT_EPISODE_UUID = "episode_uuid"
+
+        fun enqueue(episode: BaseEpisode, constraints: Constraints = Constraints.NONE, context: Context) {
+            if (episode !is PodcastEpisode) {
+                return
+            }
+
+            LogBuffer.i(LogBuffer.TAG_BACKGROUND_TASKS, "$TASK_NAME - enqueued ${episode.uuid}")
+            val cacheShowNotesData = Data.Builder()
+                .putString(INPUT_PODCAST_UUID, episode.podcastUuid)
+                .build()
+            val workRequest = OneTimeWorkRequestBuilder<UpdateShowNotesTask>()
+                .setInputData(cacheShowNotesData)
+                .addTag(episode.uuid)
+                .apply {
+                    if (constraints != Constraints.NONE) {
+                        setConstraints(constraints)
+                    }
+                }
+                .build()
+            WorkManager.getInstance(context).beginUniqueWork(TASK_NAME, ExistingWorkPolicy.APPEND, workRequest).enqueue()
+        }
     }
 
     private val podcastUuid = inputData.getString(INPUT_PODCAST_UUID) ?: ""
-    private val episodeUuid = inputData.getString(INPUT_EPISODE_UUID) ?: ""
 
     override suspend fun doWork(): Result {
+        info("Worker started - podcast: $podcastUuid")
+        val startTime = SystemClock.elapsedRealtime()
         return try {
-            showNotesManager.downloadShowNotes(podcastUuid = podcastUuid, episodeUuid = episodeUuid)
+            showNotesManager.downloadToCacheShowNotes(podcastUuid = podcastUuid)
+            info("Worker completed - took ${SystemClock.elapsedRealtime() - startTime} ms")
             Result.success()
         } catch (e: Exception) {
-            LogBuffer.e(LogBuffer.TAG_BACKGROUND_TASKS, e, "Failed to update show notes")
-            if (e !is HttpException) {
-                SentryHelper.recordException(e)
-            }
+            info("Worker failed - took ${SystemClock.elapsedRealtime() - startTime} ms")
+            val logPriority = if (e is HttpException) Log.INFO else Log.ERROR
```

Comment:
I wonder if there is a better way to do this. Maybe I should add a common way to add network exceptions which doesn't report to Sentry. 🤔